atspi-common/src/registry.rs

```diff
@@ -0,0 +1,252 @@
+);
+
+impl_from_user_facing_type_for_event_enum!(EventListenerRegisteredEvent, Event::Listener);
+
+impl_try_from_event_for_user_facing_type!(
+	EventListenerRegisteredEvent,
+	EventListenerEvents::Registered,
+	Event::Listener
+);
+
+event_test_cases!(EventListenerRegisteredEvent, Explicit);
+
+// The registry string cannot be found in upstrream at-spi2-core.
+impl_member_interface_registry_string_and_match_rule_for_event!(
+	EventListenerRegisteredEvent,
+	"EventListenerRegistered",
+	"org.a11y.atspi.Registry",
+	"registry:event-listener-registered",
+	"type='signal',interface='org.a11y.atspi.Registry',member='EventListenerRegistered'"
+);
+
+/// Signal type emitted by `EventListenerRegistered` and `EventListenerDeregistered` signals,
+/// which belong to the `Registry` interface, implemented by the registry-daemon.
+#[validate(signal: "EventListenerRegistered")]
+#[derive(Debug, Clone, Serialize, Deserialize, Type, PartialEq, Eq, Hash)]
+pub struct EventListeners {
+	pub bus_name: OwnedUniqueName,
+	// TODO: `path` should be a `zvariant::ObjectPath` but that requires changing the signature with an attribute
+	// and `Serialize`/`Deserialize` impls.
+	pub path: String,
+}
+
+impl Default for EventListeners {
+	fn default() -> Self {
+		Self {
+			bus_name: UniqueName::from_static_str_unchecked(":0.0").into(),
+			path: String::from("/org/a11y/atspi/accessible/null"),
+		}
+	}
+}
+
+#[cfg(test)]
+mod event_listener_tests {
+	use super::*;
+
+	#[test]
+	fn test_event_listener_default_no_panic() {
+		let el = EventListeners::default();
+		assert_eq!(el.bus_name.as_str(), ":0.0");
+		assert_eq!(el.path.as_str(), "/org/a11y/atspi/accessible/null");
+	}
+}
+
+pub mod socket {
+	//! This module contains the event that is emitted by the registry daemon's `Socket` interface.
+
+	use crate::events::{DBusInterface, DBusMatchRule, DBusMember, RegistryEventString};
+	use crate::{events::MessageConversion, AtspiError, EventProperties, ObjectRef};
+	use zbus::message::Body as DbusBody;
+
+	#[cfg(feature = "zbus")]
+	use serde::{Deserialize, Serialize};
+	#[cfg(feature = "zbus")]
+	use zbus::message::Header;
+
+	/// An event that is emitted when the registry daemon has started.
+	///
+	/// The accessibility registry emits this signal early during startup,
+	/// when it has registered with the DBus daemon and is available for
+	/// calls from applications.
+	#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Default, Eq, Hash)]
+	pub struct AvailableEvent {
+		/// The emitting [`ObjectRef`].
+		pub item: ObjectRef,
+
+		/// The [`ObjectRef`] for the Registry's root object.
+		pub socket: ObjectRef,
+	}
+
+	impl_event_type_properties_for_event!(AvailableEvent);
+
+	event_test_cases!(AvailableEvent, Explicit);
+
+	// We cannot really register at Registry for this event, as it is emitted by the Registry itself at early startup.
+	// So, we do not have a registry event string for this event.
+	impl_member_interface_registry_string_and_match_rule_for_event!(
```

Comment:
Good, catch. It's unconditionally emitted in `at-spi2-core/registryd/registry.c:1437`. Remove "really" in "cannot really register"; mention it's unconditional.